Description: IPv6 traffic class support
 Here is patch that adds traffic class support for IPv6.
 It's the same that TOS byte for IPv4, commandline key is the same -S.
From: http://sourceforge.net/p/iperf/patches/25/
Date: 2012-10-02

Index: iperf-2.0.13+dfsg1/src/PerfSocket.cpp
===================================================================
--- iperf-2.0.13+dfsg1.orig/src/PerfSocket.cpp	2019-01-23 18:26:36.335695523 +0100
+++ iperf-2.0.13+dfsg1/src/PerfSocket.cpp	2019-01-23 18:26:36.327695462 +0100
@@ -156,9 +156,19 @@
     if ( inSettings->mTOS > 0 ) {
         int  tos = inSettings->mTOS;
         Socklen_t len = sizeof(tos);
-        int rc = setsockopt( inSettings->mSock, IPPROTO_IP, IP_TOS,
-                             (char*) &tos, len );
-        WARN_errno( rc == SOCKET_ERROR, "setsockopt IP_TOS" );
+        if ( !isIPV6( inSettings ) )
+        {
+           // for IPv4
+           int rc = setsockopt( inSettings->mSock, IPPROTO_IP, IP_TOS,
+                                (char*) &tos, len );
+           WARN_errno( rc == SOCKET_ERROR, "setsockopt IP_TOS" );
+        } else
+        {
+           // for IPv6 (traffic class)
+           int rc = setsockopt( inSettings->mSock, IPPROTO_IPV6, IP_TOS,
+                                (char*) &tos, len );
+           WARN_errno( rc == SOCKET_ERROR, "setsockopt IPV6_TCLASS" );
+        }
     }
 #endif
 
